src/haystack/encoding/zinc/decode/scalar/date_time.rs

```diff
@@ -102,11 +102,23 @@ pub(crate) fn parse_datetime<R: Read>(scanner: &mut Scanner<R>) -> Result<DateTi
         match fixed_offset {
```

Comment:
Nested `match` expressions involving `Option` and `Result` are usually avoided thanks to Rust's syntactic sugar. It also reduces code repetition. Something like this would yield the same result.
```suggestion
        fixed_offset
            .map_or_else(
                || make_date_time_with_tz(&utc.with_timezone(&Utc.fix()), &tz),
                |offset| {
                    offset
                        .with_ymd_and_hms(
                            date.year(),
                            date.month(),
                            date.day(),
                            time.hour(),
                            time.minute(),
                            time.second(),
                        )
                        .single()
                        .and_then(|dt| dt.with_nanosecond(time.nanosecond()))
                        .ok_or_else(|| String::from("Invalid date time."))
                        .and_then(|fixed| make_date_time_with_tz(&fixed, &tz))
                },
            )
            .map(DateTime::from)
            .or_else(|err| scanner.make_generic_err(&err))
```

What do you think?

Reply:
Looks better, applied it.